### PR TITLE
finished edit inline mode

### PR DESCRIPTION
**Pull request by ohadsahar**

### CODE REVIEW

File: core/components/TaskForm/TaskForm.tsx [hunk 1]
```diff
@@ -2,27 +2,36 @@ import { Strings } from "@/constants/strings";
...
 
 interface TaskFormProps {
   id?: string;
   taskName: string;
   status: TaskStatusType;
-  editMode: boolean;
+  handleEditClick: Function;
 }
 
 const TaskForm = (task: TaskFormProps) => {
```
Comment (amitmarko) [inline core/components/TaskForm/TaskForm.tsx]:
```suggestion
const TaskForm = ({id, taskName, status, handleEditClick}): TaskFormProps) => {
```

File: core/components/TaskForm/TaskForm.tsx [hunk 2]
```diff
@@ -2,27 +2,36 @@ import { Strings } from "@/constants/strings";
...
 
   return (
-    <FormWrapper onSubmit={(event) => onSubmit(event)}>
+    <FormWrapper
+      onSubmit={(event) => {
+        if (task.handleEditClick) {
+          task.handleEditClick();
+        }
+        onSubmit(event);
+      }}
```
Comment (amitmarko) [inline core/components/TaskForm/TaskForm.tsx]: Move to callback function

File: core/components/TasksList/TasksList.tsx [hunk 3]
```diff
@@ -66,26 +70,27 @@ const TasksList = ({
...
-              />
+            {task.id !== editTaskId ? (
+              <div onClick={() => setEditTaskId(task.id)}>
+                <Task
+                  showCheck={listName !== TaskStatusType.COMPLETED ?? false}
+                  showPause={
+                    (listName === TaskStatusType.COMPLETED ||
+                      listName === TaskStatusType.CREATED) ??
+                    false
+                  }
```
Comment (amitmarko) [inline core/components/TasksList/TasksList.tsx]:
```suggestion
                  showPause={[TaskStatusType.COMPLETED, TaskStatusType.CREATED].includes(listName)}
```

File: shared/InputField/input_field.tsx [hunk 4]
```diff
@@ -31,7 +33,12 @@ const AppInputField = ({
         {!isEditable ? (
           <FaPlus onClick={() => onSubmit()} />
         ) : (
-          <FaCheck onClick={() => onSubmit()} />
+          <FaCheck
+            onClick={() => {
+              onSubmit();
+              handleEditClick();
+            }}
+          />
```
Comment (amitmarko) [inline shared/InputField/input_field.tsx]: Move to callback

File: core/components/TaskForm/TaskForm.tsx (hunk 1 above)
Comment (ohadsahar) [reply core/components/TaskForm/TaskForm.tsx]: Done

File: core/components/TaskForm/TaskForm.tsx (hunk 2 above)
Comment (ohadsahar) [reply core/components/TaskForm/TaskForm.tsx]: Done

File: core/components/TasksList/TasksList.tsx (hunk 3 above)
Comment (ohadsahar) [reply core/components/TasksList/TasksList.tsx]: Done

File: shared/InputField/input_field.tsx (hunk 4 above)
Comment (ohadsahar) [reply shared/InputField/input_field.tsx]: To custom hook? or just create a new function that includes both of them

File: core/components/TaskForm/hooks/useSubmitForm.ts [hunk 5]
```diff
@@ -36,14 +38,20 @@ export const useSubmitForm = (task: any) => {
...
+    const task: TaskProps = {
+      id: id ?? "",
+      taskName: taskValue.trim(),
+      status,
+    };
+    dispatch(editTask(task) as any);
   };
 
-  return { onSubmit, taskValue, onChangeTaskname, isError };
+  const handlePressedEnter = (event: any, handleEditClick: Function) => {
```
Comment (amitmarko) [inline core/components/TaskForm/hooks/useSubmitForm.ts:49]: add useCallback

File: core/components/TaskForm/hooks/useSubmitForm.ts [hunk 6]
```diff
@@ -36,14 +38,20 @@ export const useSubmitForm = (task: any) => {
   };
 
   const handleEditTask = () => {
```
Comment (amitmarko) [inline core/components/TaskForm/hooks/useSubmitForm.ts:40]: add useCallback

File: core/components/TaskForm/hooks/useSubmitForm.ts [hunk 7]
```diff
@@ -20,7 +22,7 @@ export const useSubmitForm = (task: any) => {
   const onSubmit = (event?: any) => {
```
Comment (amitmarko) [inline core/components/TaskForm/hooks/useSubmitForm.ts:22]: add useCallback

File: shared/InputField/input_field.tsx [hunk 8]
```diff
@@ -8,14 +8,21 @@ interface InputFieldProps {
...
 }
 
 const AppInputField = ({
   onChange,
   onSubmit,
   value,
   isEditable,
+  handleEditClick,
 }: InputFieldProps) => {
+  const submitChanges = () => {
```
Comment (amitmarko) [inline shared/InputField/input_field.tsx:21]: add useCallback

File: shared/InputField/input_field.tsx [hunk 9]
```diff
@@ -31,7 +38,11 @@ const AppInputField = ({
         {!isEditable ? (
           <FaPlus onClick={() => onSubmit()} />
         ) : (
-          <FaCheck onClick={() => onSubmit()} />
+          <FaCheck
+            onClick={() => {
+              submitChanges();
+            }}
+          />
```
Comment (amitmarko) [inline shared/InputField/input_field.tsx:45]:
```suggestion
            onClick={submitChanges}
          />
```

Comment (ohadsahar) [reply shared/InputField/input_field.tsx:45]: Done

**Pull request merged**